src/compas_timber/model/model.py

```diff
@@ -123,6 +123,105 @@ def element_by_guid(self, guid):
+            The element to add to the group.
+        name : str, optional
+            The name of the group to add the element to. If not provided, the element's name is used.
+
+        Returns
+        -------
+        :class:`~compas_model.models.GroupNode`
+            The group node containing the element.
+
+        Raises
+        ------
+        ValueError
+            If the group name is not provided and the element has no name.
+            If a group with same name already exists in the model.
+
+        Examples
+        --------
+        >>> from compas_timber.elements import Beam, Wall
+        >>> from compas_timber.model import TimberModel
+        >>> model = TimberModel()
+        >>> wall1_group = model.add_group_element(Wall(5000, 200, 3000, name="wall1"))
+        >>> beam_a = Beam(Frame.worldXY(), 100, 200, 300)
+        >>> model.add_element(beam_a, parent=wall1_group)
+        >>> model.has_group("wall1")
+        True
+
+        """
+        # type: (TimberElement, str) -> GroupNode
+        group_name = name or element.name
+
+        if not element.is_group_element:
+            raise ValueError("Element {} is not a group element.".format(element))
+
+        if not group_name:
+            raise ValueError("Group name must be provided or group element must have a name.")
+
+        if self.has_group(group_name):
+            raise ValueError("Group {} already exists in model.".format(group_name))
+
+        group_node = self.add_group(group_name)
+        self.add_element(element, parent=group_node)
+        return group_node
+
+    def has_group(self, group_name):
+        # type: (str) -> bool
+        """Check if a group with `group_name` exists in the model.
+
+        TODO: upstream this to compas_model
+
+        Parameters
+        ----------
+        group_name : str
+            The name of the group to query.
+
+        Returns
+        -------
+        bool
+            True if the group exists in the model.
+        """
+        return group_name in (group.name for group in self._tree.groups)
+
+    def get_elements_in_group(self, group_name, filter_=None):
+        """Get all elements in a group with `group_name`.
+
+        TODO: upstream this to compas_model
+
+        Parameters
+        ----------
+        group_name : str
+            The name of the group to query.
+        filter_ : callable, optional
+            A filter function to apply to the elements.
+
+        Returns
+        -------
+        Generator[:class:`~compas_model.elements.Element`]
+            A generator of elements in the group.
+
+        """
+        # type: (str, Optional[callable]) -> Generator[Element, None, None]
+        if not self.has_group(group_name):
+            raise ValueError("Group {} not found in model.".format(group_name))
+
+        filter_ = filter_ or (lambda _: True)
+
+        group = next((group for group in self._tree.groups if group.name == group_name))
```

Comment:
here you use the `next` function in case there is more than one group with the same group name? If yes, can't a potential group name conflict create more issues elsewhere?

Reply:
no, multiple groups with the same name are not allowed.
see https://github.com/gramaziokohler/compas_timber/pull/320/files#diff-2231cbd16043765cae13ca138761d868f5b2bc3e264227a237f2dab618ee29d8R172

I'm using `next` here because I expect only one result from the inner expression (which returns a generator).

This is instead of doing something really verbose like
```python3
result_group = None
for group in self._tree.groups:
    if group.name == group_name:
        result_group = group
        break
```
just more compact and clean.

I could also do something like:
```python3
group = [group for group in self._tree.groups if group.name == group_name][0]
```
but I find that ugly